include/dfu/dfu_target.h

```diff
@@ -36,6 +36,7 @@ struct dfu_target {
 	int (*init)(size_t file_size, dfu_target_callback_t cb);
 	int (*offset_get)(size_t *offset);
 	int (*write)(const void *const buf, size_t len);
+	int (*erase)(void);
```

Comment:
I would let it be asynchronous and take a callback instead of block. Erasing memory takes a long time.

Reply:
Agreed that's probably best so that people can decide to make it blocking or not.